File: Central Programming League/Central Programming League Q2R3/B.cpp
#include <bits/stdc++.h>
using namespace std;

string mahoa(string s)
{
    string data = {};
    data = data + s[0];
    char current = s[0];
    for (int i = 1; i < s.length(); i++)
    {
        if (s[i] != current)
        {
            data += s[i];
            current = s[i];
        }
    }
    int j = 0;
    vector<int> num;
    int stack = 0;
    for (int i = 0; i < s.length(); i++)
    {
        if (s[i] == data[j])
        {
            stack += 1;
        }
        else
        {
            num.push_back(stack);
            stack = 1;
            j += 1;
        }
    }
    num.push_back(stack);
    string res = "";
    for (int i = 0; i < num.size(); i++)
    {
        res += to_string(num[i]) + data[i];
    }
    return res;
}

int main()
{
    int n;
    cin >> n;
    string s;
    cin >> s;
    for (int i = 0; i < n; i++)
    {
        s = mahoa(s);
    }
    cout << s;
}

File: Central Programming League/Central Programming League Q2R3/B.py

def mahoa(s):
    s = str(s)
    data = [s[0]]
    current = s[0]
    for i in range(1, len(s)):
        if s[i] != current:
            data.append(s[i])
            current = s[i]
    j = 0

    num = []
    stack = 0
    for i in range(len(s)):
        if s[i] == data[j]:
            stack += 1
        else:
            num.append(stack)
            stack = 1
            j += 1
    num.append(str(stack))

    res = ""
    for i in range(len(num)):
        res += str(num[i])+data[i]
    return res


n = int(input())  # So lan ma hoa
s = input()
for i in range(n):
    s = mahoa(s)
print(s)
